src/main/scala/com/mongodb/spark/rdd/partitioner/MongoPaginateBySizePartitioner.scala

```diff
@@ -63,22 +64,35 @@ class MongoPaginateBySizePartitioner extends MongoPartitioner with MongoPaginati
    * @return the partitions
    */
   override def partitions(connector: MongoConnector, readConfig: ReadConfig, pipeline: Array[BsonDocument]): Array[MongoPartition] = {
-
     Try(PartitionerHelper.collStats(connector, readConfig)) match {
       case Success(results) =>
         val partitionerOptions = readConfig.partitionerOptions.map(kv => (kv._1.toLowerCase, kv._2))
         val partitionKey = partitionerOptions.getOrElse(partitionKeyProperty, DefaultPartitionKey)
         val partitionSizeInBytes = partitionerOptions.getOrElse(partitionSizeMBProperty, DefaultPartitionSizeMB).toInt * 1024 * 1024
-        val count = results.getNumber("count").longValue()
+
         val avgObjSizeInBytes = results.get("avgObjSize", new BsonInt64(0)).asNumber().longValue()
         val numDocumentsPerPartition: Int = math.floor(partitionSizeInBytes.toFloat / avgObjSizeInBytes).toInt
 
-        val rightHandBoundaries = numDocumentsPerPartition >= count match {
-          case true => Seq.empty[BsonValue]
-          case false =>
-            calculatePartitions(connector, readConfig, partitionKey, count, numDocumentsPerPartition)
+        val matchQuery = PartitionerHelper.matchQuery(pipeline)
```

Comment:
Similar to the `MongoPaginateByCountPartitioner` changes,  this partitioner now uses a `count` query with the `matchQuery` to get the size of the results rather than using the `collstats` `count` value if there is a query.

If there is a `$match` filter the last partition is changed to be inclusive of the last value by using `setLastBoundaryToLessThanOrEqualTo`.